### DISCUSSION TITLE
Improved Multiline TRL Calibration Algorithm

### DISCUSSION THREAD

**Discussion by jhillairet** [scikit-rf/scikit-rf#737] [Ideas]
As seen in the mailing list, @ZiadHatab has developed an improved Multiline TRL Calibration Algorithm, available at:
https://github.com/ZiadHatab/multiline-trl-calibration

more information are available here:
https://graz.pure.elsevier.com/en/activities/improving-the-reliability-of-the-multiline-trl-calibration-algori

Eventually, this calibration method could be integrated to scikit-rf? What do you think @ZiadHatab?

**Comment by ZiadHatab** [comment]
@jhillairet 
To be honest, you can do whatever you want. No objections from my side. I personally stopped using NIST mTRL since long ago. From my experience, I know it can be difficult to convince people that NIST mTRL is not the best, because they are NIST! I tried to argue in my paper mathematically to why my method is better and gave examples.

I think it is a good idea to included it in skrf, hopefully more people can test for themselves and see the differences. To be honest, if you design your mTRL kit correctly, both methods should give the same result. It is only when you design your standards on the edge of "correct", then NIST mTRL suffers because of the enforced linearity and line selection procedure.


P.S.: if anyone wants a copy of my paper, please contact me at z.hatab@tugraz.at. Because of IEEE copyright, I can't make the paper online available. For some reason the guys at ARFTG still have not made 98th proceedings available in IEEExplore (https://ieeexplore.ieee.org/xpl/conhome/1000049/all-proceedings).
